examples/bri-3/src/bri/identity/bpiSubjects/agents/bpiSubjects.agent.ts

```diff
@@ -42,13 +42,20 @@ export class BpiSubjectAgent {
   public async fetchUpdateCandidateAndThrowIfUpdateValidationFails(
     id: string,
   ): Promise<BpiSubject> {
-    const bpiSubjectToUpdate = await this.storageAgent.getBpiSubjectById(id);
+    const bpiSubjectToUpdate: BpiSubject =
+      await this.storageAgent.getBpiSubjectById(id);
 
     if (!bpiSubjectToUpdate) {
       throw new NotFoundException(NOT_FOUND_ERR_MESSAGE);
     }
 
-    return bpiSubjectToUpdate;
+    return new BpiSubject(
```

Comment:
can we use mapper here as well?

Reply:
Mapper needs a little modification to invoke the constructors dynamically. I'll make that change in a separate PR.